gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -171,46 +166,272 @@ def _get_prior_measurements_in_world_frame(
+            # Only add an inlier camera-track measurements if the camera has other camera-camera inliers.
+            if (C(cam_id), L(track_id)) in inliers and cam_id in inlier_cameras:
+                inlier_w_i2Ui1_dict_tracks[(track_id, cam_id)] = w_i2Ui1_dict_tracks[(track_id, cam_id)]
+
+        return inlier_w_i2Ui1_dict, inlier_w_i2Ui1_dict_tracks, inlier_cameras
+
+    def __get_initial_values(self, wTi_initial: List[Optional[PosePrior]]) -> gtsam.Values:
         """Converts translations from a list of absolute poses to gtsam.Values for initialization.
 
         Args:
-            wTi_initial: List of
+            wTi_initial: List of absolute poses.
+
+        Returns:
+            gtsam.Values containing initial translations (uses symbols for camera index).
         """
         initial = gtsam.Values()
         for i, wTi in enumerate(wTi_initial):
             if wTi is not None:
-                initial.insertPoint3(i, wTi.value.translation())
+                initial.insertPoint3(C(i), wTi.value.translation())
         return initial
 
+    def _select_tracks_for_averaging_by_measurements(
+        self,
+        tracks: List[SfmTrack2d],
+        valid_cameras: Set[int],
+        intrinsics: List[Optional[gtsfm_types.CALIBRATION_TYPE]],
+        measurements_per_camera=TRACKS_MEASUREMENTS_PER_CAMERA,
+    ) -> List[SfmTrack2d]:
+        """Removes bad tracks and selects the longest ones until all cameras see `measurements_per_camera` tracks.
+
+        Bad tracks are those that have fewer than 3 measurements from valid_cameras.
+        Sorts the remaining tracks in descending order, then selects the longest ones until all cameras see at least
+         `measurements_per_camera` tracks.
+        This is based on 1dsfm's implementation here:
+        https://github.com/wilsonkl/SfM_Init/blob/fd012ef93462b8623e8d65fa0c6fa95b32270a3c/sfminit/transproblem.py#L235
+
+        Args:
+            tracks: List of all input tracks.
+            valid_cameras: Set of valid camera indices (these have direction measurements and valid rotations).
+            intrinsics: List of camera intrinsics.
+            measurements_per_camera: Number of track direction measurements that need to be observed by each camera.
+
+        Returns:
+            List of tracks to use for averaging.
+        """
+        filtered_tracks = []
+        valid_cameras_with_intrinsics = set([c for c in valid_cameras if intrinsics[c] is not None])
+        for track in tracks:
+            valid_cameras_track = track.select_for_cameras(camera_idxs=valid_cameras_with_intrinsics)
+            if valid_cameras_track.number_measurements() < MIN_TRACK_MEASUREMENTS_FOR_AVERAGING:
+                continue
+            filtered_tracks.append(valid_cameras_track)
+
+        tracks_subset = []
+
+        remaining_cover = {c: measurements_per_camera for c in valid_cameras_with_intrinsics}
```

Comment:
Could we add an inline comment explaining what `remaining_cover` means? didn't quite follow